services/book.services.js

```diff
@@ -0,0 +1,81 @@
+const logger = require("../logs/logger");
+const db = require("../models");
+const ERROR_MSG = require("../utils/const");
+const { getPagination, getPagingData } = require("../utils/helper");
+const Book = db.books;
+const operator = db.Sequelize.Op;
+
+function createNewBook(
+  title,
+  isbn,
+  author,
+  createdBy,
+  updatedBy,
+  totalNumber,
+  nosAvailable
+) {
+  return Book.create({
+    title,
+    isbn,
+    author,
+    createdBy,
+    updatedBy,
+    totalNumber,
+    nosAvailable,
+  })
+    .then((book) => {
+      return book;
+    })
+    .catch((err) => {
+      logger.error(ERROR_MSG, err);
+      next(err);
+    });
+}
+
+function fetchAllBooks(
+  page,
+  size,
+  title,
+  author,
+  isbn,
+  createdBy,
+  isAvailable
+) {
+  const { limit, offset } = getPagination(page, size);
+  const conditions = [
+    title ? { title: { [operator.iLike]: `%${title}%` } } : null,
+    author ? { author: { [operator.iLike]: `%${author}%` } } : null,
+    isbn ? { isbn: { [operator.iLike]: `%${isbn}%` } } : null,
+    createdBy ? { createdBy: { [operator.iLike]: `%${createdBy}%` } } : null,
+    isAvailable ? { isAvailable: isAvailable } : null,
+  ];
+
+  const filteredConditions = conditions.filter(
+    (condition) => condition !== null
+  );
+
+  const query = filteredConditions.length
+    ? {
+        where: {
+          [operator.or]: filteredConditions,
```

Comment:
Do not use or. Use and. As a user, I can decide to filter with available date and title so i don't start beating around the bush.

Also, remove date like. Use date range for date. In that case, something like `between` is expected